iotdb-client/jdbc/src/main/java/org/apache/iotdb/jdbc/relational/IoTDBRelationalDatabaseMetadata.java

```diff
@@ -0,0 +1,279 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.jdbc.relational;
+
+import org.apache.iotdb.jdbc.Field;
+import org.apache.iotdb.jdbc.IoTDBAbstractDatabaseMetadata;
+import org.apache.iotdb.jdbc.IoTDBConnection;
+import org.apache.iotdb.jdbc.IoTDBJDBCResultSet;
+import org.apache.iotdb.service.rpc.thrift.IClientRPCService;
+
+import org.apache.tsfile.enums.TSDataType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class IoTDBRelationalDatabaseMetadata extends IoTDBAbstractDatabaseMetadata {
+
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(IoTDBRelationalDatabaseMetadata.class);
+
+  private static final String DATABASE_VERSION =
+      IoTDBRelationalDatabaseMetadata.class.getPackage().getImplementationVersion() != null
+          ? IoTDBRelationalDatabaseMetadata.class.getPackage().getImplementationVersion()
+          : "UNKNOWN";
+
+  public static final String SHOW_TABLES_ERROR_MSG = "Show tables error: {}";
+
+  public IoTDBRelationalDatabaseMetadata(
+      IoTDBConnection connection, IClientRPCService.Iface client, long sessionId, ZoneId zoneId) {
+    super(connection, client, sessionId, zoneId);
+  }
+
+  @Override
+  public String getDriverVersion() throws SQLException {
+    return DATABASE_VERSION;
+  }
+
+  @Override
+  public ResultSet getTables(
+      String catalog, String schemaPattern, String tableNamePattern, String[] types)
```

Comment:
`schemaPattern` and `tableNamePattern` are botn pattern, you should use like to do that filter. And the interface definition also tells the order of the result, you should order by database, table_name.

you can use information_schema.tables to do that. https://timechor.feishu.cn/docx/Mqi9dCcu2oXLo4xxZVbcn8SKnKh